usr/share/rear/format/USB/default/300_format_usb_disk.sh

```diff
@@ -1,63 +1,69 @@
-# $answer is filled by 200_check_usb_layout.sh
-if [[ "$answer" == "Yes" || "$FORCE" ]]; then
-    umount $REAL_USB_DEVICE >&8 2>&1
-
-    LogPrint "Repartition $RAW_USB_DEVICE"
-
-    echo "Yes" | parted -s $RAW_USB_DEVICE mklabel msdos >&2
-    StopIfError "Could not create msdos partitioning"
-
-    if [[ "$EFI" == "y" ]]; then
-        LogPrint "The --efi toggle was used with format - make an EFI bootable USB disk"
-
-        # Prompt user for size of EFI partition on USB disk
-        # Pressing Enter (\n) will use default value from default.conf
-        echo -n "Please enter size of EFI partition on USB device in MB [default ${USB_UEFI_PART_SIZE} MB]: "
-        read efi_part_size
-
-        # Check if user entered unsigned integer larger than 0
-        if [[ "${efi_part_size}" =~ ^[0-9]+$ && ${efi_part_size} > 0 ]]; then
-            USB_UEFI_PART_SIZE=${efi_part_size}
-            LogPrint "Creating EFI partition on USB device with size ${USB_UEFI_PART_SIZE} MB."
-        # We did not read anything, used defaults
-        elif [[ -z ${efi_part_size} ]]; then
-            LogPrint "Creating EFI partition on USB device with default size ${USB_UEFI_PART_SIZE} MB."
-        # User input was not correct ...
-        else
-            Error "Bad input for EFI partition size."
-        fi
-
-        echo "Yes" | parted -s $RAW_USB_DEVICE -- mklabel gpt mkpart primary 0 ${USB_UEFI_PART_SIZE}Mib mkpart primary ${USB_UEFI_PART_SIZE}Mib 100% >&2
-
-        StopIfError "Could not create primary partitions on '$REAL_USB_DEVICE'"
-        # partition 1 is the ESP (vfat partition) on which EFI/BOOT/BOOTX86.EFI resides
-        ParNr=2
+
+# $USB_format_answer is filled by 200_check_usb_layout.sh
+[[ "$USB_format_answer" == "Yes" || "$FORCE" ]] || return 0
+
+umount $REAL_USB_DEVICE &>/dev/null
+
+LogPrint "Repartitioning '$RAW_USB_DEVICE'"
+
+echo "Yes" | parted -s $RAW_USB_DEVICE mklabel msdos >&2
+StopIfError "Could not create initial msdos partition table on '$REAL_USB_DEVICE'"
+
+test "$USB_DEVICE_FILESYSTEM_PERCENTAGE" || USB_DEVICE_FILESYSTEM_PERCENTAGE="100"
+
+if [[ "$EFI" == "y" ]]; then
+    LogPrint "The --efi toggle was used with format - make an EFI bootable device '$REAL_USB_DEVICE'"
+
+    # Prompt user for size of EFI partition on USB disk
+    # Pressing Enter (\n) will use USB_UEFI_PART_SIZE (default in default.conf) or fallback here:
+    test "$USB_UEFI_PART_SIZE" || USB_UEFI_PART_SIZE="100"
```

Comment:
Actually it makes a lot of sense.

First and foremost it is fail-safe programming.
I need a non-empty USB_UEFI_PART_SIZE
and I ensure that gegardless of any blind assumptions
that the right stuff is set in default.conf and/or that
the user has the right stuff set in his local.conf.

Secondly I like to be prepared to have an empty
default in default.conf.
When the user has specified USB_UEFI_PART_SIZE
in his local.conf I do not need to ask him interactively
for the value.
This is because I think and ReaR workflow must be able
to run unattended - provided the user has in advance
specified all needed information in his local.conf.